srai/loaders/osm_way_loader/osm_way_loader.py

```diff
@@ -0,0 +1,373 @@
+"""
+OSM Way loader.
+
+This module contains osm loader implementation for ways based on OSMnx.
+"""
+import logging
+from enum import Enum
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import shapely.geometry as shpg
+from functional import seq
+from tqdm.auto import tqdm
+
+from srai.constants import WGS84_CRS
+from srai.exceptions import LoadedDataIsEmptyException
+from srai.utils._optional import import_optional_dependencies
+
+from . import constants
+
+logger = logging.getLogger(__name__)
+
+
+class NetworkType(str, Enum):
+    """
+    Type of the street network.
+
+    See [1] for more details.
+
+    References:
+        [1] https://osmnx.readthedocs.io/en/stable/osmnx.html#osmnx.graph.graph_from_place
+    """
+
+    ALL_PRIVATE = "all_private"
+    ALL = "all"
+    BIKE = "bike"
+    DRIVE = "drive"
+    DRIVE_SERVICE = "drive_service"
+    WALK = "walk"
+
+
+class OSMWayLoader:
+    """
+    OSMWayLoader downloads road infrastructure from OSM.
+
+    OSMWayLoader loader is a wrapper for the `osmnx.graph_from_polygon()`
+    and `osmnx.graph_to_gdfs()` that simplifies obtaining the road infrastructure data
+    from OpenStreetMap. As the OSM data is often noisy, it can also take an opinionated approach
+    on preprocessing it having standarisation in mind - e.g. converting to the same units,
+    disgarding non-wiki values and rounding them.
```

Comment:
```suggestion
    from OpenStreetMap. As the OSM data is often noisy, it can also take an opinionated approach
    to preprocessing it, with standardisation in mind - e.g. unification of units,
    discarding non-wiki values and rounding them.
```